amplitude_python_sdk/v2/client.py

```diff
@@ -81,3 +89,75 @@ def batch_upload_events(
         """  # pylint: disable=line-too-long
         req_data = self._construct_event_request_data(events, options)
         return self._log_events_internal(routes.BATCH_API, req_data, timeout)
+
+    @staticmethod
+    def _construct_chart_request_data(
+        annotation: ChartAnnotationsV2
+    ) -> Dict[str, Any]:
+        """
+        Convert ChartAnnotationsV2 model to dictionary payload for _chart_annotations_post request
+        """
+        req_data = annotation.dict(exclude_none=True)
+
+        req_data["date"] = req_data["date"].__str__()
+
+        return req_data
+
+    def _charts_annotations_post(
+        self, req_data: Dict[str, Any], timeout: int = 5,
+    ) -> requests.Response:
+        """
+        Create an annotation
+        """
+        try:
+            resp = requests.post(
+                url=self.api_endpoint_2 + routes.CHART_ANNOTATIONS_API,
+                auth=(self.api_key, self.secret_key),
+                json=req_data,
+                timeout=timeout
+            )
+            return return_or_raise(resp)
+        except requests.exceptions.Timeout as exc:
+            raise AmplitudeAPIException() from exc
+
+    def charts_annotations_upload_annotation(
+        self, charts: ChartAnnotationsV2, timeout: int = 5
+    ) -> requests.Response:
+        """
+        Parses ChartAnnotationsV2 and calls post function
+        """  # pylint: disable=line-too-long
+        req_data = self._construct_chart_request_data(charts)
+        return self._charts_annotations_post(req_data, timeout)
+
+    def chart_annotations_get_all(
+        self, timeout: int = 5,
+    ) -> requests.Response:
+        """
+        Get all annotations
+        """
+        try:
+            resp = requests.get(
+                url=self.api_endpoint_2 + routes.CHART_ANNOTATIONS_API,
+                auth=(self.api_key, self.secret_key),
+                timeout=timeout
+            )
+            return return_or_raise(resp)
+        except requests.exceptions.Timeout as exc:
+            raise AmplitudeAPIException() from exc
+
+    def chart_annotations_get_by_id(
```

Comment:
I think this can pretty easily be unified with `chart_annotations_get_all` into one `chart_annotations_get` function that takes an optional ID. If the ID is not provided we can default to returning all annotations.

Reply:
Done